src/components/PeopleFilter/PeopleFilter.tsx

```diff
@@ -0,0 +1,138 @@
+      } else if (Array.isArray(value)) {
+        searchParams.delete(key);
+        value.forEach(part => {
+          searchParams.append(key, part);
+        });
+      } else {
+        searchParams.set(key, value);
+      }
+    });
+
+    setSearchParams(searchParams);
+  };
+
+  const handleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    updateSearchParams({ query: event.target.value || null });
+  };
+
+  const handleResetAllFilters = () => {
+    updateSearchParams({
+      query: null,
+      centuries: null,
+      sex: null,
+    });
+  };
+
+  return (
+    <nav className="panel">
+      <p className="panel-heading">Filters</p>
+
+      <p className="panel-tabs" data-cy="SexFilter">
+        <SearchLink
+          className={cn({ 'is-active': !sex })}
+          params={{ sex: null }}
+        >
+          All
+        </SearchLink>
+
+        <SearchLink
+          className={cn({ 'is-active': sex === 'm' })}
+          params={{ sex: 'm' }}
+        >
+          Male
+        </SearchLink>
+
+        <SearchLink
+          className={cn({ 'is-active': sex === 'f' })}
+          params={{ sex: 'f' }}
+        >
+          Female
+        </SearchLink>
+      </p>
+
+      <div className="panel-block">
+        <p className="control has-icons-left">
+          <input
+            data-cy="NameFilter"
+            type="search"
+            value={query}
+            className="input"
+            placeholder="Search"
+            onChange={handleInput}
+          />
+
+          <span className="icon is-left">
+            <i className="fas fa-search" aria-hidden="true" />
+          </span>
+        </p>
+      </div>
+
+      <div className="panel-block">
+        <div className="level is-flex-grow-1 is-mobile" data-cy="CenturyFilter">
+          <div className="level-left">
+            {centuriesValues.map(century => (
+              <button
+                key={century}
+                type="button"
+                data-cy="century"
+                className={cn('button mr-1', {
+                  'is-info': centuries.includes(century),
+                })}
+                onClick={() => {
+                  updateSearchParams({
+                    centuries: centuries.includes(century)
+                      ? centuries.filter(c => c !== century)
+                      : [...centuries, century],
+                  });
+                }}
+              >
+                {century}
+              </button>
+            ))}
+          </div>
+
+          <div className="level-right ml-4">
+            <button
+              type="button"
+              data-cy="centuryALL"
+              className="button is-success is-outlined"
+              onClick={() => {
+                updateSearchParams({ centuries: null });
+              }}
+            >
+              All
+            </button>
+          </div>
+        </div>
+      </div>
+
+      <div className="panel-block">
+        <button
```

Comment:
same here

Reply:
If you will put a Link instead of a button, you don't need onClick function, just send a customer to people page and it automatically resets everything

Reply:
🔥